app/controllers/super_admin/wifi_user_searches_controller.rb

```diff
@@ -1,13 +1,11 @@
 class SuperAdmin::WifiUserSearchesController < SuperAdminController
   def show
     @form = SearchForm.new(search_term_params)
-  end
 
-  def create
-    @form = SearchForm.new(search_term_params)
-    @form_valid = @form.valid?
-    @wifi_user = @form_valid ? WifiUser.search(@form.search_term) : nil
-    render :show
+    @wifi_user = if search_term_params.present?
```

Comment:
I'm not sure about having just one 'show' method handling everything. The first time the page is shown the form should not be validated because the input is going to be invalid (the search term is blank) and when the page is submitted, validation should happen. This logic is contained in the controller logic but it isn't that clear in my opinion.

Also, using a ' GET' method here allows this page to be cached, but  if I search for a user, delete the user and then search again with the same string then the user might appear to be still there if the page was retrieved from cache?